src/foam/core/AbstractClassPropertyInfo.java

```diff
@@ -0,0 +1,10 @@
+package foam.core;
+
+import foam.lib.json.Outputter;
+
+public abstract class AbstractClassPropertyInfo extends AbstractObjectPropertyInfo {
+  @Override
+  public void toJSON(Outputter outputter, Object value) {
+    outputter.outputClassInfoValue((ClassInfo) value);
```

Comment:
This should just be outputString(((ClassInfo)value).getId()) to be consistent with the JS should it not?

Reply:
Oh nvm, I see what you did.  I don't think we really need to split outputClassInfo and outputClassInfoValue.  The only reason this was done for dates is that how we actually output a date value differs depending on the outputter outputReadableDates flag, but we have no such flag for ClassInfos, the ClassInfo value is always output the same way.

Reply:
The outputString and outputClassInfo methods of Outputter class have "protected" access modifiers thus cannot be called from AbstractClassPropertyInfo. 

The Outputter class also has a public output method and AbstractPropertyInfo, the ancestor of this class (AbstractClassPropertyInfo), already have toJSON method that calls outputter.output(value) so having AbstractClassPropertyInfo seems to not have any benefits.

Also on server-side, I wonder whether we should store the long form { "class": "__Class__", ... } or short form (class name as string) of Class objects in the journal files.

Reply:
"The outputString and outputClassInfo methods of Outputter" making those public rather than adding a new outputClassInfoValue method is simpler.

"Also on server-side, I wonder whether we should store the long form { "class": "__Class__", ... } or short form (class name as string) of Class objects in the journal files."

"In the journal files" is a special case we shouldn't consider directly.  When outputting a Class property it should output as a plain string, when outputting  a class in some other context like a generic Object property it will be `{ "class": "__Class__", ...`

So in the journals it will typically just be the Class name, but it could be the long form depending on context.

Reply:
Removed outputClassInfoValue in https://github.com/foam-framework/foam2/pull/2022/commits/28bc409f12ec9ca3497c2d7dd4ac022be2684d6a.